src/Federation/Repository/DeliveryQueueItem.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Federation\Repository;
+
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Federation\Collection;
+use Friendica\Federation\Entity;
+use Friendica\Federation\Factory;
+use Friendica\Util\DateTimeFormat;
+use Psr\Log\LoggerInterface;
+
+final class DeliveryQueueItem extends \Friendica\BaseRepository
+{
+	static $table_name = 'delivery-queue';
+
+	public function __construct(Database $database, LoggerInterface $logger, Factory\DeliveryQueueItem $factory)
+	{
+		parent::__construct($database, $logger, $factory);
+	}
+
+	public function selectByServerId(int $gsid, int $maxFailedCount): Collection\DeliveryQueueItems
+	{
+		$rows = $this->db->selectToArray(
+			static::$table_name,
+			[],
+			["`gsid` = ? AND `failed` < ?", $gsid, $maxFailedCount],
+			['order' => ['created']]
+		);
+
+		$Entities = new Collection\DeliveryQueueItems();
+		foreach ($rows as $fields) {
+			$Entities[] = $this->factory->createFromTableRow($fields);
+		}
+
+		return $Entities;
+	}
+
+	public function selectAggregateByServerId(): Collection\DeliveryQueueAggregates
+	{
+		$Entities = new Collection\DeliveryQueueAggregates();
+
+		$deliveries = $this->db->p("SELECT `gsid`, MAX(`failed`) AS `failed` FROM " . DBA::buildTableString([self::$table_name]) . " GROUP BY `gsid` ORDER BY RAND()");
+		while ($delivery = $this->db->fetch($deliveries)) {
+			$Entities[] = new Entity\DeliveryQueueAggregate($delivery['gsid'], $delivery['failed']);
+		}
+
```

Comment:
Don't forget to close the database connection at the end.

Reply:
I saw you did in your own PR 😄

Reply:
Done.